src/cosmicqc/detection.py

```diff
@@ -0,0 +1,619 @@
+        upper_skew_threshold: float = 0.09,
+        variability_threshold: float = 0.15,
+        texture_mean_threshold: float = -0.25,
+        formfactor_mean_threshold: float = 0.78,
+        outlier_std_threshold: float = 1.0,
+    ) -> None:
+        """
+        Initializer for the ContaminationDetector class.
+
+        Args:
+            dataframe (pd.DataFrame): The input DataFrame containing the features
+                to be analyzed. Should be the output from CytoTable convert.
+            nucleus_channel_naming (str, optional): Naming convention for the nucleus
+                channel in the DataFrame. Defaults to "DNA".
+            lower_skew_threshold (float, optional): Lower threshold for Bowley's
+                skewness to flag whole-plate contamination. Defaults to -0.15.
+            upper_skew_threshold (float, optional): Upper threshold for Bowley's
+                skewness to flag partial-plate contamination. Defaults to 0.09.
+            variability_threshold (float, optional): Threshold for acceptable
+                coefficient of variation in selected features. Defaults to 0.15.
+            texture_mean_threshold (float, optional): Mean texture value threshold
+                for detecting abnormalities. Defaults to -0.25.
+            formfactor_mean_threshold (float, optional): Mean FormFactor threshold
+                for detecting nuclear shape issues. Defaults to 0.78.
+            outlier_std_threshold (float, optional): Number of standard deviations from
+                the mean used to flag outliers. Defaults to 1.0.
+        """
+        self.dataframe = dataframe
+        self.nucleus_channel_naming = nucleus_channel_naming
+        self.lower_skew_threshold = lower_skew_threshold
+        self.upper_skew_threshold = upper_skew_threshold
+        self.variability_threshold = variability_threshold
+        self.texture_mean_threshold = texture_mean_threshold
+        self.formfactor_mean_threshold = formfactor_mean_threshold
+        self.outlier_std_threshold = outlier_std_threshold
+
+        # set the features to be used for contamination detection
+        self.cyto_feature = (
+            f"Cytoplasm_Texture_InfoMeas1_{self.nucleus_channel_naming}_3_02_256"
+        )
+        self.formfactor_feature = "Nuclei_AreaShape_FormFactor"
+
+    def skewness_test_cytoplasm_texture(self) -> bool:
+        """
+        Bowley's skewness score is calculated for the cytoplasm texture
+        around the nucleus feature.
+        Thresholds are assigned to determine if there is abnormal texture around
+        the nucleus, either based on a serious negative skew
+        (indicating a whole plate issue) or a positive skew
+        (indicating a partial plate issue).
+        Thresholds are set based on findings from multiple experiments
+        (NF1 & CFReT, see citation file).
+
+        Reference: Bowley, A. L. (1901): Elements of Statistics, P.S. King and Son, Laondon.
```

Comment:
Nice - consider also adding this to the `CITATION.cff` references.

Reply:
I think since I can't really find an article for this with all the authors, I am going to leave the reference here as I don't think I can fill out the citation file correctly.